src/components/popover/popover.jsx

```diff
@@ -10,43 +10,33 @@ import IconButton from '../icon-button/icon-button.jsx';
 import iconClear from '../../svgs/clear.svg';
 import styles from './popover.css';
 import useFlush from '../../hooks/use-flush.js';
+import useMounted from '../../hooks/use-mounted.js';
 
 const Popover = forwardRef(function Popover(props, ref) {
-  const [open, setOpen] = useState(props.startOpen);
+  const [open, setOpen] = useState(false);
   const closeIconRef = useRef();
   const backupRef = useRef();
   const sectionRef = useRef();
   const selfRef = ref || backupRef;
   const { flush } = useFlush({ ref: sectionRef, initialDirection: props.flush, autoflush: props.autoflush, open });
+  const mounted = useMounted();
 
-  const onFocusIn = (event) => {
-    if (open && event.target instanceof Node && !sectionRef.current.contains(event.target)) {
+  function onBlur(event) {
+    // Target is set to `null` when losing focus to a non-interactive element (e.g. text)
+    const defaultShouldClose = event.relatedTarget === null || !sectionRef.current.contains(event.relatedTarget);
+    if (props.shouldClose(event) && defaultShouldClose) {
```

Comment:
1. This is pedantic, but this isn't really a `default`, it's just internal logic. `default` to me implies that it can be overridden (like default props). Related, I wonder if we can use that excuse to find a name that explains why and not what, and roll the explanation from the comment into the name of the variable, like `hasFocusShifted` or something along those lines
2. Kinda dumb, but would it make sense to swap the order of these two conditions? `defaultShouldClose` seems like it'll always be fast to evaluate, but since `shouldClose` comes from `props`, it could require a large calculation (depending on the consumer of the design system). We might get random short-circuiting benefits from an order change.

Reply:
1. Yeah, maybe just `shouldClose`? 
2. That's interesting, yeah, I can switch the order.